docs/adr/0046-monorepo-split.md

```diff
@@ -0,0 +1,143 @@
+
+SubPlat has considered moving to a separate repository for the maintenance and deployment of Subscription Platform-specific functionality.
+
+**Pain points that led to considering a split:**
+
+* Slow CI (often ≥ \~20 minutes).
+* Legacy TS+JS in `auth-server` with wide use of typecasts and older testing methodologies.
+* Coupled release process: FxA and SubPlat are forced to tag and deploy together.
+* Dependency upgrades are painful across the repo.
+* Consistency of code style between the two teams varies.
+
+**Work already done as of writing this ADR:**
+
+* New SubPlat work lives under Nx (`apps/*`, `libs/*`) starting with `payments-next` for SubPlat 3.0.
+  * This helps decouple the dependence between SubPlat and FxA.
+* Nx remote caching backed by S3 to improve CI runtimes when modifying non-FxA code.
+* Reduced interdependence between FxA and SubPlat with SubPlat 3.0
+  * Removed sub-accounts.
+  * Very limited interaction directly with FxA accounts tables.
+
+**Current issues:**
+
+* CI is still not great.
+  * CI runtime varies, depending on what you touch.
+  * If functional tests have to spin up, they require a stack start.
+* Significant SubPlat logic remains in `auth-server`.
+  * All webhook processing code.
+  * All emailing logic.
+* Cross‑team style drift still happens.
+* SubPlat still reads from the FxA accounts table in a few limited places.
+
+**Question:** Should SubPlat split into its own repository, stay as-is, or remain in the monorepo while formalizing the improvements already underway?
+
+## Decision Drivers
+
+* Release independence for SubPlat.
+* CI performance and developer velocity.
+* Migration cost and risk.
+* Ability to share code (`libs/shared/*`) and infra (CI, image building).
+* Ongoing maintenance burden (dependency upgrades, code organization).
+* Reduction of legacy coupling to `auth-server`.
+
+## Considered Options
+
+A. **Split SubPlat into its own repo and abandon the monorepo.**
+B. **Status quo, do nothing**
+C. **Stay in the monorepo and implement the improvements outlined above.**
+
+## Decision Outcome
+
+**Chosen option: C — Stay in the monorepo and implement targeted improvements.**
+
+We believe this best fits the current state: we already have `payments-next` in Nx, remote caching is in place, and nothing actually prevents an independent release cadence for SubPlat when coexisting in the monorepo. We keep the benefits of shared libraries and shared CI/devops while focusing effort on extracting remaining SubPlat logic from `auth-server` into `payments-api`, and on test modernization, instead of spending that effort on a risky+costly repo split.
+
+### Positive Consequences
+
+* Preserve shared code under `libs/shared/*` and reduce duplication.
+* Keep using shared CI, devops, and image-building pipelines.
+* Avoid large one‑time migration cost; focus energy on extracting SubPlat code from `auth-server`.
+* SubPlat can immediately run an independent release cadence for `payments-next`.
```

Comment:
Is there any plan to break out separate functional (I'm specifically thinking of smoke tests here) for subplat. That could potentially help our release cadence as well.

Reply:
@dschom They do not exist at this time.